python/doc/source/numpy.rst

```diff
@@ -0,0 +1,77 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _numpy_interop:
+
+Using PyArrow with NumPy
+========================
+
+PyArrow allows converting back and forth from
+`NumPy <https://www.numpy.org/>`_ arrays to Arrow :ref:`Arrays <data.array>`.
+
+NumPy to Arrow
+--------------
+
+To convert a NumPy array to Arrow, one can simply call the :func:`pyarrow.array`
+factory function.  This function will always make a copy of the input data,
+as Arrow has stricter alignment requirements than NumPy.
```

Comment:
`pyarrow.array` is zero copy if the NumPy array is contiguous and contains numbers, otherwise (non-contiguous boolean, string, objects, etc.) it copies

Reply:
Uh, it is indeed. That is weird. Does it ignore alignment issues?

Reply:
Yes, it does probably ignore them. We should make an issue if that is really the case...

Reply:
```python
>>> import pyarrow as pa
>>> import numpy as np
>>> data = np.arange(10)
>>> arr = pa.array(data[1:])
>>> data[1] = 42
>>> arr
<pyarrow.lib.Int64Array object at 0x7f4205290818>
[
  42,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9
]
```

Reply:
It isn't a bug. If you send this data on the wire or write it to Plasma it will be aligned and padded at that point. There's no need to assert padding or alignment requirements for buffers until it is actually needed. For SIMD, padding matters more than alignment

Reply:
Hmm, I see.